### ISSUE TITLE
Improve containsFields utility function to accept mixed generic types

### ISSUE THREAD

**Issue by zeljkobekcic** [datalbry/connector-sdk#24]
# Background

Currently the utility function `Record.containsFields` only accepts only generic type. Currently, there is a case where I have mixed generics types.

# Example Usecase

```kotlin

var document = ... // some generic document
var expectedFields = listOf(
    GenericField(name = "id", value = 1), // GenericField<Int>
    GenericField(name = "name", value = "name"), // GenericField<String>
)
assertTrue(document.contiansFields(expectedFields))
```

with the current version of the function `containsFields` this is not possible. The compiler will error out with a Type mismatch in this case. When the change is applied to the signature this works as expected. 

# Where

:arrow_right:  https://github.com/datalbry/connector-sdk/tree/improvement/issue-24/containsFields-refactoring

# Awaited

- [ ] Change the function signature accordingly

**Comment by Lacritz** [comment]
Hi @zeljkobekcic,
please provide further information. I don't understand what changes you would like to see in the sdk.

**Comment by zeljkobekcic** [comment]
Merged in #25 and therefore closing it
